File: src/main/java/dev/blachut/svelte/lang/psi/SvelteTypes.java
package dev.blachut.svelte.lang.psi;

import com.intellij.psi.tree.IElementType;

public interface SvelteTypes {
    IElementType CODE_FRAGMENT = new SvelteElementType("CODE_FRAGMENT");
    IElementType HTML_FRAGMENT = new SvelteElementType("HTML_FRAGMENT");

    IElementType START_MUSTACHE = new SvelteElementType("START_MUSTACHE");
    IElementType START_MUSTACHE_TEMP = new SvelteElementType("START_MUSTACHE_TEMP");
    IElementType END_MUSTACHE = new SvelteElementType("END_MUSTACHE");

    IElementType TEMP_PREFIX = new SvelteElementType("TEMP_PREFIX");
    IElementType LAZY_IF = new SvelteElementType("LAZY_IF");
    IElementType LAZY_ELSE = new SvelteElementType("LAZY_ELSE");
    IElementType LAZY_EACH = new SvelteElementType("LAZY_EACH");
    IElementType LAZY_AWAIT = new SvelteElementType("LAZY_AWAIT");
    IElementType LAZY_THEN = new SvelteElementType("LAZY_THEN");
    IElementType LAZY_CATCH = new SvelteElementType("LAZY_CATCH");

    IElementType HASH = new SvelteElementType("HASH");
    IElementType COLON = new SvelteElementType("COLON");
    IElementType SLASH = new SvelteElementType("SLASH");
    IElementType AT = new SvelteElementType("AT");
}

File: src/main/java/dev/blachut/svelte/lang/psi/SvelteTypes.kt
package dev.blachut.svelte.lang.psi

import com.intellij.psi.tree.IElementType

object SvelteTypes {
    @JvmField
    val CODE_FRAGMENT: IElementType = SvelteElementType("CODE_FRAGMENT")
    @JvmField
    val HTML_FRAGMENT: IElementType = SvelteElementType("HTML_FRAGMENT")

    @JvmField
    val START_MUSTACHE: IElementType = SvelteElementType("START_MUSTACHE")
    @JvmField
    val START_MUSTACHE_TEMP: IElementType = SvelteElementType("START_MUSTACHE_TEMP")
    @JvmField
    val END_MUSTACHE: IElementType = SvelteElementType("END_MUSTACHE")

    @JvmField
    val TEMP_PREFIX: IElementType = SvelteElementType("TEMP_PREFIX")
    @JvmField
    val LAZY_IF: IElementType = SvelteElementType("LAZY_IF")
    @JvmField
    val LAZY_ELSE: IElementType = SvelteElementType("LAZY_ELSE")
    @JvmField
    val LAZY_EACH: IElementType = SvelteElementType("LAZY_EACH")
    @JvmField
    val LAZY_AWAIT: IElementType = SvelteElementType("LAZY_AWAIT")
    @JvmField
    val LAZY_THEN: IElementType = SvelteElementType("LAZY_THEN")
    @JvmField
    val LAZY_CATCH: IElementType = SvelteElementType("LAZY_CATCH")

    @JvmField
    val HASH: IElementType = SvelteElementType("HASH")
    @JvmField
    val COLON: IElementType = SvelteElementType("COLON")
    @JvmField
    val SLASH: IElementType = SvelteElementType("SLASH")
    @JvmField
    val AT: IElementType = SvelteElementType("AT")
}
